src/client/app/components/LineChartComponent.tsx

```diff
@@ -0,0 +1,152 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as moment from 'moment';
+import { PlotRelayoutEvent } from 'plotly.js';
+import * as React from 'react';
+import Plot from 'react-plotly.js';
+import { TimeInterval } from '../../../common/TimeInterval';
+import { readingsApi } from '../redux/api/readingsApi';
+import { useAppDispatch, useAppSelector } from '../redux/reduxHooks';
+import { selectLineChartQueryArgs } from '../redux/selectors/chartQuerySelectors';
+import { selectLineChartDeps, selectPlotlyGroupData, selectPlotlyMeterData } from '../redux/selectors/lineChartSelectors';
+import { selectLineUnitLabel } from '../redux/selectors/plotlyDataSelectors';
+import { graphSlice } from '../redux/slices/graphSlice';
+import { LineReadings } from '../types/readings';
+import translate from '../utils/translate';
+import LogoSpinner from './LogoSpinner';
+import { selectSelectedLanguage } from '../redux/slices/appStateSlice';
+import locales from '../types/locales';
+
+// Stable reference for when there is not data.
+const stableEmptyReadings: LineReadings = {}
```

Comment:
It is intentionally left outside. Since it is declared outside it will retain a stable object reference. This is there to prevent unnecessary re-rendering when initial data is in flight. Putting it inside will create a new reference each render which defeats its purpose. These stable references could be moved to a different location , or maybe wrapped on a selector if desired. This pattern will show up in a handful of places so we need to decide on this sooner than later.

Reply:
Clarified comment